src/Streamly/Prelude.hs

```diff
@@ -332,11 +335,26 @@ head m =
         yield a _ = return (Just a)
     in (runStream (toStream m)) Nothing stop yield
 
+-- | Extract all but the first element of the stream, if any.
+tail :: (Streaming t, Monad m) => t m a -> m (Maybe (t m a))
+tail m =
+    let stop             = return Nothing
+        yield _ Nothing  = return Nothing
```

Comment:
On second thoughts, I think the previous version of this was the correct one. We should differentiate between empty tail and no tail. That's the behavior for lists as well.